File: 20220810/모범답안/20220810_05.js/numberCheck.py
def countDigit(number, d, K):
    power = 10 ** d
    nextPower = power * 10
    right = number % power

    down = number - number % nextPower
    up = down + nextPower

    digit = (number / power) % 10
    if digit < K:
        return down / 10
    elif digit == K:
        return down / 10 + right + 1
    else:
        return up / 10


def solution(N, K):
    count = 0

    for i in range(len(str(N))):
        count += countDigit (N, i, K)
    return count

File: 20220810/모범답안/20220810_05.js/numberCheck.js
function countDigit(number, d, k){
    var power = 10 ** d;
    var nextPower = power * 10;
    var right = number % power;
  
    var down = number - parseInt(number % nextPower);
    var up = down + nextPower;
  
    digit = parseInt(number / power) % 10;
  
    if(digit < k){
      return parseInt(down / 10);
    } else if(digit == k){
      return parseInt(down / 10) + right + 1;
    } else{
      return parseInt(up / 10);
    }
  }
  
  function solution(N, K){
    var count = 0;
  
    for(var i = 0; i < N.toString().length; i++){
      count += countDigit(N, i, K);
    }
  
    return count;
}  